ddtrace/contrib/aiokafka/patch.py

```diff
@@ -0,0 +1,248 @@
+        span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
+        span.set_tag_str(kafkax.TOPIC, topic)
+        span.set_tag_str(kafkax.MESSAGE_KEY, key)
+
+        span.set_tag(kafkax.PARTITION, partition)
+        span.set_tag_str(kafkax.TOMBSTONE, str(value is None))
+        span.set_tag(SPAN_MEASURED_KEY)
+        if instance._dd_bootstrap_servers is not None:
+            span.set_tag_str(kafkax.HOST_LIST, ",".join(instance._dd_bootstrap_servers))
+        rate = config.kafka.get_analytics_sample_rate()
+        if rate is not None:
+            span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, rate)
+
+        # inject headers with Datadog tags if trace propagation is enabled
+        if config.kafka.distributed_tracing_enabled:
+            # inject headers with Datadog tags:
+            additional_headers = {}
+            Propagator.inject(span.context, additional_headers)
+            for header, value in additional_headers.items():
+                headers.append((header, value.encode("utf-8")))
+
+        args, kwargs = set_argument_value(args, kwargs, 5, "headers", headers, override_unset=True)
+        return await func(*args, **kwargs)
+
+
+async def traced_send_and_wait(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+    result = await func(*args, **kwargs)
+    core.dispatch("aiokafka.produce.completed", (result,))
+    return result
+
+
+async def traced_getone(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    # we must get start time now since execute before starting a span in order to get distributed context
+    # if it exists
+    start_ns = time_ns()
+    err = None
+    result = None
+    try:
+        result = await func(*args, **kwargs)
+    except Exception as e:
+        err = e
+        raise err
+    finally:
+        _instrument_message(result, pin, start_ns, instance, err)
+    return result
+
+
+async def traced_getmany(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    # we must get start time now since execute before starting a span in order to get distributed context
+    # if it exists
+    start_ns = time_ns()
+    err = None
+    result = None
+    try:
+        result = await func(*args, **kwargs)
+    except Exception as e:
+        err = e
+        raise err
+    finally:
+        for tp, records in result.items():
+            for record in records:
+                _instrument_message(record, pin, start_ns, instance, err)
+    return result
+
+
+def _instrument_message(message, pin, start_ns, instance, err):
+    ctx = None
+    if message is not None and config.kafka.distributed_tracing_enabled and message.headers:
+        ctx = Propagator.extract(dict(message.headers))
+    with pin.tracer.start_span(
```

Comment:
Replacing this with `core.context_with_data` would clarify the separation of concerns between the Tracing product and the Kafka integration.

Reply:
@emmettbutler I've reworked a bit of the code to make use of the `core.context_with_data`, see the last commit.
Please let me know if this aligning more with how it should be done.

I've also disabled tracing on some functions for now until the basics are sorted out.
And I disabled the test for distributed tracing and DSM as well for now.